bzt/bza.py

```diff
@@ -0,0 +1,542 @@
+        """ Returns first item of non-empty list or None """
+        if len(self):
+            return self[0]
+
+        return None
+
+    def __getattr__(self, name):
+        def call_list_items(*args, **kwargs):
+            res = BZAObjectsList()
+            for item in self:
+                method = getattr(item, name)
+                chunk = method(*args, **kwargs)
+                if not isinstance(chunk, BZAObjectsList):
+                    msg = "%s.%s() must return BZAObjectsList, but returned %s"
+                    raise TypeError(msg % (type(item).__name__, name, type(chunk).__name__))
+                res += chunk
+
+            # logging.debug("%s[%s]: %s", name, len(res), json.dumps(res, indent=True))
+            return res
+
+        return call_list_items
+
+
+# ================================= Entities =================================
+
+class User(BZAObject):
+    def ping(self):
+        """ Quick check if we can access the service """
+        self._request(self.address + '/api/v4/web/version')
+
+    def accounts(self):
+        """
+        :rtype: BZAObjectsList[Account]
+        """
+        res = self._request(self.address + '/api/v4/accounts')
+        return BZAObjectsList([Account(self, x) for x in res['result']])
+
+    def fetch(self):
+        res = self._request(self.address + '/api/v4/user')
+        self.update(res)
+        return self
+
+    def available_locations(self, include_harbors=False):
+        if 'locations' not in self:
+            self.fetch()
+
+        locations = {}
+        for loc in self['locations']:
+            loc_id = str(loc['id'])
+            if loc_id.startswith('harbor-') and not include_harbors:
+                continue
+            locations[str(loc['id'])] = loc
+        return locations
+
+    def collection_draft(self, name, taurus_config, resource_files):
+        if resource_files:
+            draft_id = "taurus_%s" % id(self.token)
+            self.upload_collection_resources(resource_files, draft_id)
+            taurus_config.merge({"dataFiles": {"draftId": draft_id}})
+
+        collection_draft = self._import_config(taurus_config)
+        collection_draft['name'] = name
+        return collection_draft
+
+    def _import_config(self, config):
+        url = self.address + "/api/v4/multi-tests/taurus-import"
+        resp = self._request(url, data=config, method="POST")
+        return resp['result']
+
+    def upload_collection_resources(self, resource_files, draft_id):
+        url = self.address + "/api/v4/web/elfinder/%s" % draft_id
+        body = MultiPartForm()
+        body.add_field("cmd", "upload")
+        body.add_field("target", "s1_Lw")
+        body.add_field('folder', 'drafts')
+
+        for rfile in resource_files:
+            body.add_file('upload[]', rfile)
+
+        hdr = {"Content-Type": str(body.get_content_type())}
+        resp = self._request(url, body.form_as_bytes(), headers=hdr)
+        if "error" in resp:
```

Comment:
if-clause won't be performed as we look for 'error' key in _request() and raise exception if it found